app/repositories/favourites.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import textwrap
+from datetime import datetime
+from typing import Any
+from typing import TypedDict
+from typing import cast
+
+import app.state.services
+
+# +--------------+------------------------+------+-----+---------+----------------+
+# | Field        | Type                   | Null | Key | Default | Extra          |
+# +--------------+------------------------+------+-----+---------+----------------+
+# | userid       | int                    | NO   | PRI | NULL    |                |
+# | setid        | int                    | NO   |     | NULL    |                |
+# | created_at   | int                    | NO   |     | 0       |                |
+# +--------------+------------------------+------+-----+---------+----------------+
```

Comment:
UserID is the primary key? Doesn't this mean a user can only have a single favourite? Wtf

Reply:
i forgot setid :tf:

`primary key (userid, setid)`

Reply:
Kk, please use `desc` in the future to avoid any confusion/mis-reviews of PRs

Reply:
it slipped into the `create()` too -- the select isn't using the setid

Reply:
i'll fix -- check the diffs for some ideas perhaps